From 6f7017d51f54534d3e6d7aea3e1e481659a9f30f Mon Sep 17 00:00:00 2001
From: Zen <z@pyl.onl>
Date: Sat, 8 Feb 2025 10:34:54 -0600
Subject: [PATCH 37/47] don't run mount_fstab if a fstab is not in the
 initramfs

since config changes should not be made in the init runlevel, and even
if additional mounts were added, they would not affect the existing
fstab file, since it was already generated by generate_fstab

Signed-off-by: Zen <z@pyl.onl>
---
 src/ugrd/fs/mounts.py | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/src/ugrd/fs/mounts.py b/src/ugrd/fs/mounts.py
index 02ea90f..bf11aa9 100644
--- a/src/ugrd/fs/mounts.py
+++ b/src/ugrd/fs/mounts.py
@@ -1,5 +1,5 @@
 __author__ = "desultory"
-__version__ = "6.6.2"
+__version__ = "6.6.3"
 
 from pathlib import Path
 from re import search
@@ -734,6 +734,9 @@ def mount_fstab(self) -> list[str]:
     Keeps re-attempting with mount_timeout or rootdelay until successful.
     mount_retries sets the number of times to retry the mount, infinite otherwise.
     """
+    if not self._get_build_path("/etc/fstab").exists():
+        return self.logger.warning("No initramfs fstab found, skipping mount_fstab.")
+
     out = [
         'einfo "Attempting to mount all filesystems."',
         f"timeout=$(readvar rootdelay {self.get('mount_timeout', 1)})",
-- 
2.48.1

